packages/sdks/web-component/src/lib/descope-wc/DescopeWc.ts

```diff
@@ -636,6 +645,41 @@ class DescopeWc extends BaseDescopeWc {
           {},
         );
 
+        // Try to detect whether the tab is being throttled when running in a mobile browser, specifically on iOS.
+        // We check whether the tab seems to hidden and the polling callback was called much later than expected,
+        // in which case we allow a much shorter timeout for the polling request. The reschedule check ensures
+        // this cannot happen twice consecutively.
+        const throttled =
+          document.hidden &&
+          !rescheduled &&
+          Date.now() - scheduledAt > delay + pollingThrottleThreshold;
+        if (throttled) {
+          this.logger.debug('polling - The polling seems to be throttled');
+        }
+
+        let sdkResp: Awaited<typeof nextCall>;
+        try {
+          const timeout = throttled
+            ? pollingThrottleTimeout
+            : pollingDefaultTimeout;
+          sdkResp = await timeoutPromise(timeout, nextCall);
+        } catch (err) {
+          this.logger.warn(
```

Comment:
can the promise may reject regardless the timeout, no? if server will return non 200 response then we may arrive here

I'm not recall if sever returns an  error throws or just return `resp.ok == false`

(if the former, then I think we can get infinite loop here)

Reply:
I believe @nirgur changed the networking layer so that it always catches thrown errors and converts them to `resp.error`, etc.